graylog2-server/src/main/java/org/graylog2/streams/StreamRouterEngine.java

```diff
@@ -137,33 +143,63 @@ public String getFingerprint() {
      * @return the list of matching streams
      */
     public List<Stream> match(Message message) {
```

Comment:
Finally changing the method return type to `Set<Stream>` would make sense now, I guess. This would also avoid having to copy the matching result at the end of this method.